### PR TITLE
Support delete

### PR DESCRIPTION
**Pull request by iamlinjunhong**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2408 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by daviszhen — APPROVED** [review]

File: proto/plan.proto [hunk 1]
```diff
@@ -316,98 +316,89 @@ message UpdateList {
...
-	Expr limit			= 15;
-	Expr offset			= 16;
-	TableDef table_def	= 17;
-	ObjectRef obj_ref	= 18;
+	Expr limit = 15;
+	Expr offset = 16;
+	TableDef table_def = 17;
+	ObjectRef obj_ref = 18;
 	RowsetData rowset_data = 19;
-	string extra_options   = 20;
```
Comment (ouyuanning) [inline proto/plan.proto:400]: extra_options was added by Dr Tian。
I  don't known if this options has any other means。
I think you can just leave it here.
You can add your options in DeleteInfo

File: pkg/sql/plan2/build_delete.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func buildDelete(stmt *tree.Delete, ctx CompilerContext) (*Plan, error) {
 		return nil, errors.New(errno.SyntaxErrororAccessRuleViolation, "cannot find delete table")
 	}
 
-	// build the stmt of select
+	// build the projection of select
 	var projectExprs tree.SelectExprs
```
Comment (ouyuanning) [inline pkg/sql/plan2/build_delete.go:42]: seams projectExprs is an unused variable?
then buildProjectionFromExpr , buildProjectionFromExpr are unused function?

File: proto/plan.proto (hunk 1 above)
Comment (iamlinjunhong) [reply proto/plan.proto:400]: This field has not been deleted. You can take a closer look.

File: pkg/sql/plan2/build_delete.go (hunk 2 above)
Comment (iamlinjunhong) [reply pkg/sql/plan2/build_delete.go:42]: projectExprs is used to check whether the primary key is a filter condition

**Review by ouyuanning — APPROVED** [review]
ok

**Pull request merged**